src/Validation/Validator.php

```diff
@@ -433,10 +440,30 @@ public function remove($field, $rule = null)
      */
     public function requirePresence($field, $mode = true, $message = null)
     {
-        $this->field($field)->isPresenceRequired($mode);
-        if ($message) {
-            $this->_presenceMessages[$field] = $message;
+        if (!is_array($field)) {
+            $field = [
+                $field => [
+                    'mode' => $mode,
+                    'message' => $message
+                ]
+            ];
         }
+
+        foreach ($field as $fieldName => $setting) {
+            if (is_string($setting)) {
```

Comment:
so 
`if (is_string($setting) && is_int($fieldName)) {`
isnt that a bit redundant check?